tests/unit/test_arg_recorder.py

```diff
@@ -136,16 +136,32 @@ def test_node_table(self):
         assert num_lines(nt) == self.arg.num_nodes + 1
         # one line for each node and one for header
 
-    def test_add_samples(self):
-        samples = [0, 1]
-        self.arg.add_samples(samples, 1.0)
+    def test_sample_table(self):
         out = six.StringIO()
         self.arg.dump_sample_table(out)
         print(out.getvalue())
-        assert num_lines(out.getvalue()) == len(samples) + 1  # for header
+        assert num_lines(out.getvalue()) == 1  # no samples
 
-    def test_sample_table(self):
+
+class TestARGrecorderGoodSampledPopulation():
+    def setup_method(self):
+        '''setup for all tests in class'''
+        self.arg = ARGrecorder()
+        self.arg.add_individual(2, time=3.0)
+        self.arg.add_individual(3, time=3.0)
+        self.arg.add_individual(4, time=1.0)
+        self.arg.add_record(left=0.5, right=1.0, parent=3, children=(4, ))
+        self.arg.add_record(left=0.0, right=0.5, parent=2, children=(4, ))
+        self.arg.add_individual(1, time=0.0)
+        self.arg.add_individual(0, time=0.0)
+        self.samples = [3, 4]
+        self.arg.add_samples(samples=self.samples, length=1.0)
```

Comment:
@petrelharp any insight on why this fails to create some samples in `arg`? It's very similar to what you get when running [the `wf` test](https://github.com/ashander/ftprime/blob/fe002ed47bf4cf0af2b64579572a4098790aa434/tests/test_arg_recorder_with_wf.py#L22-L23) with 2 samples (with `=5` replaced by `=2`), which works fine but this doesn't.

You can see test output here: https://circleci.com/gh/ashander/ftprime/58?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link